src/trace-agent.js

```diff
@@ -61,22 +61,6 @@ function TraceAgent(config, logger) {
 }
 
 /**
- * Halts this agent and unpatches any patched modules.
- */
-TraceAgent.prototype.stop = function() {
```

Comment:
Having `stop()` in `test/hooks/common.js` still requires support on the `src` side, such as exposing functionality to reset the reference to `agent` in `index.js`. No matter what there will have to be something exposed that is testing-only, and I went with `forceNewAgent_` because it (1) doesn't introduce extra methods and (2) is uniform with what we did in the Debug Agent.